### PR TITLE
Add transaction handler RPC for Mysticeti fastpath

### PR DESCRIPTION
**Pull request by aschran**
This will not function until consensus adapter/handler side is implemented.

### CODE REVIEW

File: crates/sui-core/src/authority_server.rs
Comment (mwtian) [inline crates/sui-core/src/authority_server.rs:477]: It seems more refactoring will be needed to support soft bundle. Most of the logic here should be shared between handle_transaction_v2 and handle_soft_bundle_transactions_v2 handlers. Or we can take the opportunity to support batched transaction submission with a single handler, which should cover the soft bundle use case as well.

File: crates/sui-network/build.rs
Comment (lxfind) [reply crates/sui-network/build.rs:39]: Why wouldn't there be a signature with the effects?

File: crates/sui-types/src/messages_grpc.rs
Comment (lxfind) [inline crates/sui-types/src/messages_grpc.rs:223]: I wonder if we should give it a new schema of names.
Having HandleTransaction in the request type but HandleCertificate in the response type is confusing to me.

File: crates/sui-core/src/authority.rs
Comment (lxfind) [inline crates/sui-core/src/authority.rs:1012]: This doesn't seem to be the right naming anymore since we are only doing various checks here.

**Review by lxfind — COMMENTED** [review]
How does this look like with https://github.com/MystenLabs/sui/pull/19320?

File: crates/sui-network/build.rs
Comment (mwtian) [reply crates/sui-network/build.rs:39]: Because this handler is for fastpath, which does not produce BLS signatures per effects which is needed for `SignedTransactionEffects` included in `HandleCertificateResponseV3`.

Comment (lxfind) [reply crates/sui-network/build.rs:39]: I might be missing some context, but why doesn't it produce a signature per effects?

Comment (mwtian) [reply crates/sui-network/build.rs:39]: During the discussions we had previously, it was determined that effects certificate was unnecessary at the beginning. So not producing the effects BLS signatures to save CPU seems to be the right choice.

File: crates/sui-core/src/authority.rs
Comment (aschran) [reply crates/sui-core/src/authority.rs:1012]: Do you have an alternate suggestion for name?

I think actually the existing `handle_transaction` was poorly named and should have been called `sign_transaction`. But for the new one I'm adding, `handle` is more apt because we really are handling the full transaction e2e: sending an unsigned tx through consensus, executing it, and returning the effects. I didn't think temporary confusion of naming overload would be too bad, bc eventually we plan to remove the old version.

File: crates/sui-types/src/messages_grpc.rs
Comment (aschran) [reply crates/sui-types/src/messages_grpc.rs:223]: I can make a new response type, was just trying to save time thinking the output would be the same, but clearly that's wrong anyway if as Mingwei says we won't be able to sign the effects

**Comment by aschran** [comment]
> How does this look like with #19320?

I don't think anything changes here? We aren't acquiring locks for the tx in the RPC handler anymore, that will need to be done on the consensus handler side

File: crates/sui-network/build.rs
Comment (aschran) [reply crates/sui-network/build.rs:39]: How does the caller verify that a quorum of validators has approved the result without the effects sig?

Comment (aschran) [reply crates/sui-network/build.rs:39]: I guess they couldn't have done that with just one authority sig anyway but I'm curious how it works right now.

File: crates/sui-core/src/authority_server.rs
Comment (aschran) [reply crates/sui-core/src/authority_server.rs:477]: Yes, I was not originally planning to support soft bundle for the first prototype. do you think that's worth adding to this PR?

File: crates/sui-core/src/authority.rs
Comment (aschran) [reply crates/sui-core/src/authority.rs:1043]: The locking would need to happen on the consensus handler side, no?

File: crates/sui-network/build.rs
Comment (mwtian) [reply crates/sui-network/build.rs]: Discussed with @lxfind offline. The critical thing is that we will need to support TLS for transaction handler.

Comment (aschran) [reply crates/sui-network/build.rs]: I would consider supporting TLS for tx handler pretty high priority regardless of anything to do with fast path

File: crates/sui-core/src/authority.rs
Comment (aschran) [reply crates/sui-core/src/authority.rs]: Added locking as per discussion

**Comment by aschran** [comment]
> #19320



> > How does this look like with #19320?
> 
> I don't think anything changes here? We aren't acquiring locks for the tx in the RPC handler anymore, that will need to be done on the consensus handler side

Ok this is now relevant since I added locking, commented on that PR

File: crates/sui-network/build.rs
Comment (mwtian) [reply crates/sui-network/build.rs]: Great! That would be a good follow up.

File: crates/sui-core/src/authority_server.rs
Comment (mwtian) [reply crates/sui-core/src/authority_server.rs]: As discussed, we can definitely can back and refactor later.

File: crates/sui-core/src/authority_server.rs
Comment (mwtian) [inline crates/sui-core/src/authority_server.rs:478]: Is this TODO still relevant?

**Review by mwtian — APPROVED** [review]
Lgtm but let's wait for a day or two for reviews from other folks.

File: crates/sui-core/src/authority.rs
Comment (arun-koshy) [inline crates/sui-core/src/authority.rs:1018]: handle_transaction_v2

File: crates/sui-core/src/authority.rs
Comment (arun-koshy) [inline crates/sui-core/src/authority.rs:1215]: nit: `await_transaciton_effects`

**Review by arun-koshy — APPROVED** [review]

File: crates/sui-core/src/authority_server.rs
Comment (mwtian) [inline crates/sui-core/src/authority_server.rs:563]: Btw let's guard this or the entire handler with a protocol config flag, e.g. `mysticeti_fastpath`? Otherwise we may accidentally expose production validators to dos or crash before the logic is complete.

Comment (aschran) [reply crates/sui-core/src/authority_server.rs:563]: Good idea

**Pull request merged**
